### PR TITLE
Display base power of Hidden Power/Frustration/Return in Pokemon tooltip if not max

### PR DESCRIPTION
**Pull request by scheibo**
Won't do anything without https://github.com/Zarel/Pokemon-Showdown/pull/5519 which plumbs through happiness as part of the `|request|`.

### CODE REVIEW

File: src/battle-tooltips.ts [hunk 1]
```diff
@@ -733,6 +734,14 @@ class BattleTooltips {
 		return text;
 	}
 
+	getMoveDisplay(move: Move, pokemon: ServerPokemon) {
+		if (((move.id === 'frustration' || move.id === 'return') && move.basePower !== 102) ||
+			(move.id.startsWith('hiddenpower') && this.battle.gen < 7 && move.basePower !== 70)) {
+			return `${move.name} - ${move.basePower}BP`;
```
Comment (Zarel) [inline src/battle-tooltips.ts]:
```suggestion
			return `${move.name} ${move.basePower}`;
```

File: src/battle-tooltips.ts [hunk 2]
```diff
@@ -733,6 +734,14 @@ class BattleTooltips {
 		return text;
 	}
 
+	getMoveDisplay(move: Move, pokemon: ServerPokemon) {
+		if (((move.id === 'frustration' || move.id === 'return') && move.basePower !== 102) ||
+			(move.id.startsWith('hiddenpower') && this.battle.gen < 7 && move.basePower !== 70)) {
```
Comment (Zarel) [inline src/battle-tooltips.ts]: Does this correct display the type?

Comment (Zarel) [reply src/battle-tooltips.ts]:
```
> Dex.mod('gen5').getMove('hiddenpowerflying59').name
"Hidden Power Flying"
> Dex.mod('gen5').getMove('hiddenpowerflying59').id
"hiddenpowerflying59"
```

Huh, apparently yes.

File: src/battle-tooltips.ts (hunk 1 above)
Comment (scheibo) [reply src/battle-tooltips.ts]: You like `Frustration 98 (29/32)` vs `Frustration - 98BP (29/32)` (this was one thing I brought up on the [thread](https://www.smogon.com/forums/threads/return-frustration-base-power-tooltip.3650730/))?

Comment (Zarel) [reply src/battle-tooltips.ts]: Yeah `Frustration 98 (29/32)` was always how I would have implemented it, and it matches how variable base power moves are usually talked about.

https://www.google.com/search?q="hidden+power+70"

See how people just say "Hidden Power 70" when talking about it.

Comment (scheibo) [reply src/battle-tooltips.ts]: Ack. Just wanted to make sure

**Comment by scheibo** [comment]
So did some testing (btw, the teambuilder is super obnoxious about not wanting me to have non 0 or 255 happiness, but thats besides the point), needs more work

File: src/battle-dex.ts [hunk 3]
```diff
@@ -278,7 +278,7 @@ const Dex = new class implements ModdedDex {
 			let [, hpWithType, hpPower] = /([a-z]*)([0-9]*)/.exec(id)!;
 			data = {
 				...(window.BattleMovedex[hpWithType] || {}),
-				basePower: Number(hpPower) || 60,
+				basePower: Number(hpPower) || (gen < 6 ? 70 : 60),
```
Comment (Zarel) [inline src/battle-dex.ts]: In theory, the server should send the HP power if it's 70.

Comment (Zarel) [reply src/battle-dex.ts]: Also there's no need to ask for `gen`, you can just test `this.gen`.

**Pull request closed without merging**